src/Aspire.Hosting.Elasticsearch/ElasticsearchResource.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// A resource that represents a Elasticsearch
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+public class ElasticsearchResource(string name) : ContainerResource(name), IResourceWithConnectionString
```

Comment:
Consider looking into password support by default. We should be password protecting all of our resources by default.

Reply:
@davidfowl 
Add password support in the [commit](https://github.com/dotnet/aspire/pull/4430/commits/ca9709c1a15ec37768652c4cb35be64692bd26b8).

I only used the password as a resource parameter in the constructor because it is not possible to change the built-in elastic superuser name, but I added the username as a property in the resource to have access to the username property out of class (for different authentication way in clients)